File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/tinsel/TitleSelectedMessage.scala





















// Generated on 05/03/2013 10:35:00
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait TitleSelectedMessage extends MessageDeserializer {
  val messageId = 6366

  protected trait Message extends BaseMessage with Serializable {
    val messageId = TitleSelectedMessage.messageId

    val titleId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(titleId)
    }

  }

  protected def create(titleId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val titleId = buf.readShort()

    create(titleId)
  }

}

object TitleSelectedMessage extends TitleSelectedMessage {
  case class MessageImpl(titleId: Short) extends Message
  type Target = MessageImpl

  override def create(titleId: Short) = MessageImpl(titleId)

  def apply(titleId: Short) = create(titleId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/TitleSelectedMessage.java


// Generated on 05/08/2013 19:37:58
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class TitleSelectedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6366;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short titleId;
    
    public TitleSelectedMessage() { }
    
    public TitleSelectedMessage(short titleId) {
        this.titleId = titleId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(titleId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        titleId = buf.readShort();
        if (titleId < 0)
            throw new RuntimeException("Forbidden value on titleId = " + titleId + ", it doesn't respect the following condition : titleId < 0");
    }
    
}
